x-pack/plugin/identity-provider/src/main/java/org/elasticsearch/xpack/idp/action/TransportSamlInitiateSingleSignOnAction.java

```diff
@@ -64,7 +66,8 @@ protected void doExecute(Task task, SamlInitiateSingleSignOnRequest request,
                     }
                     final SecondaryAuthentication secondaryAuthentication = SecondaryAuthentication.readFromContext(securityContext);
                     if (secondaryAuthentication == null) {
-                        listener.onFailure(new IllegalStateException("Request is missing secondary authentication"));
+                        listener.onFailure(
+                            new ElasticsearchStatusException("Request is missing secondary authentication", RestStatus.FORBIDDEN));
```

Comment:
Can we make it `ElasticsearchSecurityException` instead?

```suggestion
                            new ElasticsearchSecurityException("Request is missing secondary authentication", RestStatus.FORBIDDEN));
```